### ISSUE TITLE
Set default values for schema `url` fields

### ISSUE THREAD

**Issue by HosseinAgha** [prisma/prisma#222]
## My proposals

1. Add a syntax in Prisma Schema Language for setting a default value if env variable was empty. e.g.
```groovy
datasource db {
  provider = "postgresql"
  url = env('POSTGRES_URL') || "postgresql://dev@localhost/testdb"
}
```
2. As a better option add support for schema definition in JavaScript/Typescript (e.g. `schema.prisma.js`)

## Why I think creating a new schema language isn't a good idea?

- Please re read your article on [the problems of schema first development](https://www.prisma.io/blog/the-problems-of-schema-first-graphql-development-x1mn4cb0tyl3).
- I worry that in time you reach the same conclusion and you end up with a schema language that tries to solve everything but is not as good as a real programming language.
- I think having a real general purpose programming language at your disposal is always a good idea. (take Webpack config or many other tools for example)
- I understand that Prisma is a cross language tool (Go driver for example) but it is also written in JS and Generator and Lift need node to run. So I see no harm in having a  `schema.prisma.js` option.

## What are the use cases for these?

I’m struggling to add Prisma 2 to my development workflow.  
I have a complex NodeJS server that connects to some DBs and outside services. We try our best to keep initial configuration for new developers as simple as possible (setup everything in the first `npm run dev`).  
We use docker to initialize and run the DBs and external services and setup some default credentials for dev databases.  
In the dev server we set the same default credentials for our DB connectors so new developers don’t have to worry about DB config at all.  
The current `url` system for Prima Schema Language only accepts environment variables. It is a good thing for production but not the dev env.  
Using `.env` file is not a good option for us as each new developer should create it and add the default configs to it.

**Comment by HosseinAgha** [comment]
It is good to mention that we are currently using `photon`'s datasources argument to override the database URL like this:  
```typescript
const photon = new Photon({
      datasources: {
          db: {
            url: process.env.POSTGRES_URL || "postgresql://dev@localhost/repository",
          },
      },
});
```

**Comment by janpio** [comment]
A note: This might also be solved by adding a second param default (of fallback) to the `env()` method.

**Comment by matthewmueller** [comment]
Hi @HosseinAgha! Thanks for bringing this up, 

For **1.**, we're definitely going to add a default env argument, it'll likely look like this

```
env("POSTGRES_URL", default: "postgresql://dev@localhost/testdb")
```

For **2.**, thanks for bringing this up! You've sparked a lively debate internally about this decision. 

The reason we went with a declarative format rather than a code-first format is that we're describing the layout of your datasource, we're not providing implementations for this schema (unlike graphql's resolvers). In this sense, we're more like JSON schema or terraform than GraphQL. We also want to be able to migrate your database based on this schema file.

We do understand that you may want to simply generate database clients and bring your own migration engine, so for that we're planning to introduce the Prisma SDK to work with Prisma without generating a schema.

**Comment by blevine** [comment]
Was the default for the connection url functionality ever added?

**Comment by pantharshit00** [comment]
@blevine No, we didn't get any more requests for this so this was never pushed again up the priority classification.

**Comment by blevine** [comment]
That's too bad.  It seems to have been almost completely implemented and would have been a good enhancement.

**Comment by danablend** [comment]
I would be happy to see an implementation of this as well! :-)

**Comment by 2pmflow** [comment]
Would love to see this implemented. It would help with zero downtime migrations of database credentials.

Supporting something similar to `env("NEW_DATABASE_URL") || env("DATABASE_URL")` would let the application handle two database credentials at once.

**Comment by janpio** [comment]
How would that work @2pmflow, specifically? If it falls back to the second value if the first env var is empty, can you not also just update the only single environment variable to the new value instead?

**Comment by flockonus** [comment]
This issue is from Jul 2019; 6 years later there is still no way to way to read an env var falling back to a default value in Prisma?
Am I missing something?

**Comment by igalklebanov** [comment]
Hey 👋 

Starting from Prisma v7, schema files no longer support this field - it was moved to the new JS/TS config file, making this feature request irrelevant.
